### PR TITLE
Add support for traffic-routing spec v1alpha1

### LINKED ISSUE #33: Watch OpenSergo Label Routing CRD
To make OpenSergo support LabelRouting, we must watch the OpenSergo's CRD in control plane. And once we found the OpenSergo's CRD is changed, we will resolve the CRD to Istio RouteConfiguration object and push them to OperSergo SDK so that it can fit Istio ecosystem as well.
The work flow chart is as below:
image

### PR DESCRIPTION
**Pull request by 123liuziming**
### Describe what this PR does / why we need it
We can observe the label routing crd provided by OpenSergo, and publish it to SDK
### Does this pull request fix one issue?
Fixes #33

### Describe how you did it
see #33 

### Special notes for reviews

### CODE REVIEW

**Review by jnan806 — COMMENTED** [review]
Could you please add an issue for this PR, and add some description of your design and implementation in this PR description

File: .gitignore [hunk 1]
```diff
@@ -26,6 +26,12 @@
 .idea/**/gradle.xml
 .idea/**/libraries
 
+.idea/.gitignore
+.idea/modules.xml
+.idea/opensergo-control-plane.iml
+.idea/protoeditor.xml
+.idea/vcs.xml
+
```
Comment (jnan806) [inline .gitignore:34]: I think all files from IDE of `intelliJ IDEA` is unnecessary, so i think we can ignore them like follwing:

```
.idea/
.idea_modules/
*.iml
```

**Review by jnan806 — CHANGES_REQUESTED** [review]

File: pkg/api/v1beta1/networking/groupversion_info.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package networking
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "networking.istio.io", Version: "v1beta1"}
```
Comment (jnan806) [inline pkg/api/v1beta1/networking/groupversion_info.go:28]: Maybe need to relate with `opensergo`

File: .gitignore (hunk 1 above)
Comment (123liuziming) [reply .gitignore:34]: > I think all files from IDE of `intelliJ IDEA` is unnecessary, so i think we can ignore them like follwing:
> 
> ```
> .idea/
> .idea_modules/
> *.iml
> ```

I think so,  I will modify the .gitignore

**Review by jnan806 — COMMENTED** [review]
It's better to re-format `*.go` files by goimports,  directory `pkg/api` and directory `pkg/proto` can be ignored.

**Comment by jnan806** [comment]
rename `config/samples/istio_viretual_service_samples.yaml` with opensergo is better.

`pkg/controller/crd_transform.go` is only used for traffic, better to indicate it in the file name.

**Comment by 123liuziming** [comment]
> rename `config/samples/istio_viretual_service_samples.yaml` with opensergo is better.
> 
> `pkg/controller/crd_transform.go` is only used for traffic, better to indicate it in the file name.

Thanks! I will fix them.

**Review by jnan806 — APPROVED** [review]
LGTM.

cc @sczyh30   any other questions?

**Review by jnan806 — APPROVED** [review]

File: pkg/controller/crd_watcher.go [hunk 3]
```diff
@@ -36,6 +37,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	EXTENSION_ROUTE_FALL_BACK = "envoy.router.cluster_specifier_plugin.cluster_fallback"
```
Comment (sczyh30) [inline pkg/controller/crd_watcher.go:41]: Maybe this should be moved to `traffic_router_transform.go`?

Also please polish the naming according to the https://go.dev/doc/effective_go#names

File: pkg/main/main.go [hunk 4]
```diff
@@ -15,9 +15,8 @@
 package main
 
 import (
-	"log"
-
 	"github.com/opensergo/opensergo-control-plane"
+	"log"
```
Comment (sczyh30) [inline pkg/main/main.go:19]: Please reformat with `goimports`

File: pkg/controller/traffic_router_transform.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
+package controller
```
Comment (sczyh30) [inline pkg/controller/traffic_router_transform.go:1]: Please add our license header here.

File: pkg/api/v1alpha1/traffic/traffic_router.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package traffic
```
Comment (sczyh30) [inline pkg/api/v1alpha1/traffic/traffic_router.go:1]: Please add the license header here.

**Review by sczyh30 — CHANGES_REQUESTED** [review]

**Review by sczyh30 — COMMENTED** [review]
File name typo: `config/crd/bases/traffic.opensergi.io_trafficerouters.yaml`

**Comment by sczyh30** [comment]
The CI lint failure might be caused by low version of the linter. I'll fix it later.

**Comment by sczyh30** [comment]
cc @jnan806 @123liuziming @panxiaojun233 I'll use "squash and merge" for this PR to keep our commits tidy.

**Comment by sczyh30** [comment]
Nice work. Thanks for contributing!

**Pull request merged**